src/client.rs

```diff
@@ -353,6 +355,15 @@ impl ClientConfig {
             ..self
         }
     }
+
+    /// Configure compression for requests and responses.
+    /// It's enabled by default.
```

Comment:
Conflicted on whether to enable by default for zstd. I guess disabled would be most consistent across SDKs where we can only support gzip for now. And compression is not helpful in certain cases. So let's keep it opt-in everywhere.